twitchpopups.htm

```diff
@@ -97,59 +118,93 @@ <h1 id="popuptext"></h1>
 
             if (context.mod || (context["badges-raw"] != null && context["badges-raw"].startsWith("broadcaster"))) {
                 if (commandName.startsWith("!alert ")) {
-                    $("#popupbox").show();
-                    $("#popupbox").css({"background-color":alertBg}); 
-                    $("#popuptext").text(commandName.substr(7).toUpperCase());
+                    setText(commandName.substr(7).toUpperCase(), alertBg);
                     doAnimation();
                 } else if (commandName.startsWith("!delete")) {
                   deleteAnimation();
                 } else if (commandName.startsWith("!hotseat ")) {
                   hotSeatIsOn = true;
-                  $("#popupbox").show();
-                  hotSeatUser = commandName.substr(10).toLowerCase();
-                  $("#popupbox").css({"background-color":hotseatBg}); 
-                  $("#popuptext").text(`${hotseatEmoji}  ${hotSeatUser.toUpperCase()} IS IN THE HOTSEAT ${hotseatEmoji}`);
+                  hotSeatUser = commandName.replace(/!hotseat @?/, '').toLowerCase();
+                  setText(`${hotseatEmoji} ${hotSeatUser.toUpperCase()} IS IN THE HOTSEAT ${hotseatEmoji}`, hotseatBg);
                   doAnimation();
                 }
             }
                 
             if (hotSeatIsOn) {
                 if (context.username == hotSeatUser){
-                    $("#popuptext").text(`${hotseatEmoji} ${context['display-name']}: ${commandName} ${hotseatEmoji}`);
+                    setText(`${hotseatEmoji} ${context['display-name']}: ${commandName} ${hotseatEmoji}`, hotseatBg);
                     doHotseatAnimation();
                 }
             }
                 
     }
         
+    const setText = (text, bgColor) => {
+        popupText.innerText = text;
+        const isOverflowing = popupText.offsetWidth > popupBox.offsetWidth;
+        if (isOverflowing) {
+            popupBox.style.backgroundColor = bgColor;
+            popupText.style.backgroundColor = 'transparent';
+        } else {
+            popupText.style.backgroundColor = bgColor;
+            popupBox.style.backgroundColor = 'transparent';
+        }
+        if (isOverflowing) {
+            if (marqueeStylesheet.sheet.cssRules.length > 0) {
+                marqueeStylesheet.sheet.deleteRule(0);
+            }
+            marqueeStylesheet.sheet.insertRule(`@keyframes marquee {
+                0% {
+                    transform: translate3d(${popupBox.offsetWidth / 4}px, 0, 0);
+                    opacity: 0;
+                }
+                10% {
+                    opacity: 1;
+                }
+                90% {

```

Comment:
What's the point of keyframe at 90%? It can be removed and it will work the same.